src/store/TimeperiodStore.js

```diff
@@ -65,7 +65,7 @@ export default class TimeperiodStore {
 
             const dataSet = stx.chart.dataSet;
             if (dataSet && dataSet.length !== 0) {
-                const now = new Date();
+                const now = this.mainStore.chart.serverTime.getLocalDate();
```

Comment:
this should be changed to use UTC. Then the timezone offset below should no longer be required.

Reply:
@brucebinary  `serverTime.getUTCDate` returns an string and `dataSet[dataSet.length - 1].DT` is a date object so realy cant minus them, if thats what you meant

Reply:
I mean use UTC time here (assuming the format is correct). There's a timezone offset below, so that should no longer with necessary:

```js
// Dates are in UTC; we need to do a timezone offset
const diff = now - dataSet[dataSet.length - 1].DT + (now.getTimezoneOffset() * 60000);
```